### PR TITLE
Environment: check GD and Imagick availability & supported formats.

### PR DESCRIPTION
**Pull request by adamsilverstein**
Addresses https://github.com/WordPress/phpunit-test-runner/issues/131.

### CODE REVIEW

File: prepare.php [hunk 1]
```diff
@@ -51,6 +51,8 @@
 \$env = array(
 	'php_version'    => phpversion(),
 	'php_modules'    => array(),
+	'gd_info'        => extension_loaded( 'gd' ) ? gd_info() : 'unavailable',
+	'Imagick_info'   => extension_loaded( 'imagick' ) ? Imagick::queryFormats() : 'unavailable'
```
Comment (getsource) [inline prepare.php:55]: Should this be lowercase?

**Comment by getsource** [comment]
Thanks so much! I haven't had the chance to test yet, but this looks very close.

Due to #111 , unfortunately, this needs to be updated in two places at the moment -- both `prepare.php` and `functions.php`.

I need to resurrect that issue / patch / diff at some point (or help is welcome) to refactor and turn it into an optional SSH call for gathering environment data, but I think it's fine to go ahead and add this in the meantime.

File: prepare.php (hunk 1 above)
Comment (adamsilverstein) [reply prepare.php:55]: sure, makes sense for consistency.

**Comment by adamsilverstein** [comment]
> Due to #111 , unfortunately, this needs to be updated in two places at the moment -- both prepare.php and functions.php.

I missed that, will add there as well. Thanks for the tip.

**Comment by adamsilverstein** [comment]
Addressed feedback:
* Added to `functions.php`.
* Lower case naming.

**Review by pfefferle — APPROVED** [review]
LGTM.

**Comment by adamsilverstein** [comment]
@getsource friendly ping, this should be ready to go!

**Comment by getsource** [comment]
Thanks! Trying to test, but ran into some environment difficulties with app passwords and the reporter plugin.
Will ping back when I've been able to test.

**Review by getsource — COMMENTED** [review]
I tested, and the information gets uploaded to the reporter as expected.

A couple of notes:
- We'll need to add something to [the reporter](https://github.com/WordPress/phpunit-test-reporter/) to display this information (Edit: This does not have to happen before this gets merged, but we should probably open an Issue either now, or when it does).
- When successful, an array gets inserted for both items. Would an empty array (or not including it at all / "empty") make it easier to parse this in the reporter when "unavailable"?

**Comment by getsource** [comment]
Here are relevant locations for the reporter, for consideration of the unavailable values:
https://github.com/WordPress/phpunit-test-reporter/blob/master/parts/single-result.php#L50
https://github.com/WordPress/phpunit-test-reporter/blob/master/src/class-display.php#L337

**Comment by adamsilverstein** [comment]
I created an issue to add the data if available to the reports: https://github.com/WordPress/phpunit-test-reporter/issues/97

> When successful, an array gets inserted for both items. Would an empty array (or not including it at all / "empty") make it easier to parse this in the reporter when "unavailable"?

Good point, I hadn't considered the reporting end of things. Makes more sense for the default value to be an empty array vs. the string 'unavailable'. I will update the PR.

**Comment by adamsilverstein** [comment]
@getsource - I changed the defaults to empty array as per your suggestion. This should be good to merge.

**Comment by adamsilverstein** [comment]
@pfefferle @getsource What are the next steps to get this merged (plus https://github.com/WordPress/phpunit-test-reporter/pull/98). 

Also: once this is merged, how is deployed in the wild, when could we hope to start seeing data reported?

**Comment by getsource** [comment]
My suggested steps would be:
- Review of this PR's code
- Review of the other PR's code
- Testing both of the two together (WP.org meta environment for the latter).
- Merge of both PRs
- Create New Plugin Version for the reporter
- Create meta patch for the reporter PR (using the reporter build tools)
- Make meta ticket, with that patch attached
- Commit from the meta team to make the new version of the reporter live

I don't think this PR necessarily needs to wait for the latter, but having them work together as a proof of concept definitely would help verify/validate. 

I don't recall whether sending the data to the reporter when it isn't expecting it (if this PR is merged first) would cause validation issues with the API or not, but looking+testing to make sure it works as expected would be good before this is merged.

As far as when data is received: After the above is done, because updating the runner code is dependent on the client, I suspect some hosts will start to report right away (if they clone this repo as part of the testing process), and for others it will take an undefined amount of time until they decide to manually update.

I don't have the time at the moment to work on this, unfortunately, so help is appreciated.

**Comment by desrosj** [comment]
Just cross referencing the recent Core changeset that added this data to update pings to .org: https://core.trac.wordpress.org/changeset/53753.

**Review by javiercasares — APPROVED** [review]

**Pull request merged**